pulp_rpm/app/serializers/package.py

```diff
@@ -141,15 +141,21 @@ class PackageSerializer(SingleArtifactContentUploadSerializer, ContentChecksumSe
         read_only=True,
     )
 
+    filename = serializers.CharField(
+        help_text=_("The original filename of the package - "),
+        read_only=True,
+    )
```

Comment:
Should we expose this at all?  Filename is not intrinsic to the RPM metadata, you can change an RPM's filename and it'd be the same RPM still, and the way we deduplicate content means it's not reliable anyway.  We might have to keep `location_href` for compatibility but maybe we don't want to actively provide this info more than necessary.

Reply:
I think users would love to see the filename and search by it. It was never intrinsic or reliable  - even in pulp 2 one was able to upload package foo.rpm and set filename as bar.rpm.
As long as users don't want to shoot their feet, and i believe most of the users  don't  - they will just regularly use it without being 'inventive'.

Reply:
I'm a little more concerned with repo maintainers being too "inventive", like that SUSE issue, or the one we had with Microsoft a while back : /